rest/model/patch.go

```diff
@@ -11,30 +14,45 @@ import (
 
 // APIPatch is the model to be returned by the API whenever patches are fetched.
 type APIPatch struct {
-	Id              *string       `json:"patch_id"`
-	Description     *string       `json:"description"`
-	ProjectId       *string       `json:"project_id"`
-	Branch          *string       `json:"branch"`
-	Githash         *string       `json:"git_hash"`
-	PatchNumber     int           `json:"patch_number"`
-	Author          *string       `json:"author"`
-	Version         *string       `json:"version"`
-	Status          *string       `json:"status"`
-	CreateTime      *time.Time    `json:"create_time"`
-	StartTime       *time.Time    `json:"start_time"`
-	FinishTime      *time.Time    `json:"finish_time"`
-	Variants        []*string     `json:"builds"`
-	Tasks           []*string     `json:"tasks"`
-	VariantsTasks   []VariantTask `json:"variants_tasks"`
-	Activated       bool          `json:"activated"`
-	Alias           *string       `json:"alias,omitempty"`
-	GithubPatchData githubPatch   `json:"github_patch_data,omitempty"`
+	Id                *string           `json:"patch_id"`
+	Description       *string           `json:"description"`
+	ProjectId         *string           `json:"project_id"`
+	Branch            *string           `json:"branch"`
+	Githash           *string           `json:"git_hash"`
+	PatchNumber       int               `json:"patch_number"`
+	Author            *string           `json:"author"`
+	Version           *string           `json:"version"`
+	Status            *string           `json:"status"`
+	CreateTime        *time.Time        `json:"create_time"`
+	StartTime         *time.Time        `json:"start_time"`
+	FinishTime        *time.Time        `json:"finish_time"`
+	Variants          []*string         `json:"builds"`
+	Tasks             []*string         `json:"tasks"`
+	VariantsTasks     []VariantTask     `json:"variants_tasks"`
+	Activated         bool              `json:"activated"`
+	Alias             *string           `json:"alias,omitempty"`
+	GithubPatchData   githubPatch       `json:"github_patch_data,omitempty"`
+	ModuleCodeChanges []*APIModulePatch `json:"module_code_changes"`
 }
 type VariantTask struct {
 	Name  *string   `json:"name"`
 	Tasks []*string `json:"tasks"`
 }
 
+type FileDiff struct {
+	FileName  *string `json:"file_name"`
+	Additions int     `json:"additions"`
+	Deletions int     `json:"deletions"`
+	DiffLink  *string `json:"diff_link"`
+}
+
+type APIModulePatch struct {
+	BranchName *string     `json:"branch_name"`
+	HTMLLink   *string     `json:"html_link"`
+	RawLink    *string     `json:"raw_link"`
+	FileDiffs  []*FileDiff `json:"file_diffs"`
```

Comment:
Since you declared the contents of this array as non-nullable in the schema, this doesn't need to be a pointer does it? In general we tend to avoid arrays of pointers because they have some quirky behavior when you iterate over them